src/components/GroupBar.tsx

```diff
@@ -53,31 +53,21 @@ export default function GroupBar({
   const [isJoiningMeeting, setJoining] = useState(false);
   const [showMeetingQuotaWarning, setShowMeetingQuotaWarning] = useState(false);
 
-  const launchMeetingInNewWindow = async (groupId: string) => {
-    setJoining(true);
-    try {
-      const link = await trpc.meetings.join.mutate({ groupId: groupId });
-      if (!link) {
-        setShowMeetingQuotaWarning(true);
-      } else {
-        window.open(link, '_blank');
-      }
-    } finally {
-      setJoining(false);
-    }
-  };
-
-  const launchMeetingInSameWindow = async (groupId: string) => {
+  const launchMeeting = async (groupId: string) => {
     setJoining(true);
     try {
       const link = await trpc.meetings.join.mutate({ groupId: groupId });
       if (!link) {
         setShowMeetingQuotaWarning(true);
         setJoining(false);
       } else {
-        window.location.href = link;
-        // Time is needed for the meeting page to load.
-        setTimeout(() => setJoining(false), 5000);
+        const newWindow = window.open(link, '_blank');
+        if (!newWindow || newWindow.closed || typeof newWindow.closed === 'undefined') {
```

Comment:
link to the reference page where you get this trick. Also add a comment about the intent of this block